src/Compilers/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -1,6 +1,18 @@
 *REMOVED*Microsoft.CodeAnalysis.Operations.IEventAssignmentOperation.EventReference.get -> Microsoft.CodeAnalysis.Operations.IEventReferenceOperation
+Microsoft.CodeAnalysis.INamespaceDeclaration
+Microsoft.CodeAnalysis.INamespaceOrTypeDeclaration
+Microsoft.CodeAnalysis.INamespaceOrTypeDeclaration.Children.get -> System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.INamespaceOrTypeDeclaration>
+Microsoft.CodeAnalysis.INamespaceOrTypeDeclaration.IsNamespace.get -> bool
+Microsoft.CodeAnalysis.INamespaceOrTypeDeclaration.IsType.get -> bool
+Microsoft.CodeAnalysis.INamespaceOrTypeDeclaration.Name.get -> string
+Microsoft.CodeAnalysis.ITypeDeclaration
+Microsoft.CodeAnalysis.ITypeDeclaration.Arity.get -> int
+Microsoft.CodeAnalysis.ITypeDeclaration.Children.get -> System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.ITypeDeclaration>
+Microsoft.CodeAnalysis.ITypeDeclaration.DeclaredAccessibility.get -> Microsoft.CodeAnalysis.Accessibility
+Microsoft.CodeAnalysis.ITypeDeclaration.TypeKind.get -> Microsoft.CodeAnalysis.TypeKind
 abstract Microsoft.CodeAnalysis.Compilation.ClassifyCommonConversion(Microsoft.CodeAnalysis.ITypeSymbol source, Microsoft.CodeAnalysis.ITypeSymbol destination) -> Microsoft.CodeAnalysis.Operations.CommonConversion
 abstract Microsoft.CodeAnalysis.Compilation.ContainsSymbolsWithName(string name, Microsoft.CodeAnalysis.SymbolFilter filter = Microsoft.CodeAnalysis.SymbolFilter.TypeAndMember, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) -> bool
+abstract Microsoft.CodeAnalysis.Compilation.DeclarationRoot.get -> Microsoft.CodeAnalysis.INamespaceDeclaration
```

Comment:
>DeclarationRoot [](start = 44, length = 15)

The change looks good to me. I'm really liking the tree design (as opposed to flat list) as it fits naturally to our existing data structures.
If you get a thumbs up from Neal on API shape, the next step would be to implement compiler tests for new public APIs.

Reply:
I agree.  It's super nice since 99.9% of the time, there will be only a tiny handful of allocations post edit.  Almost the entirety of the IDeclarations will be reusable from one edit to the next